packages/react-day-picker/src/contexts/Focus/FocusContext.tsx

```diff
@@ -106,96 +101,49 @@ export function FocusProvider({
     switchMonth(down, numberOfMonths);
   };
 
-  const focusFirstDayOfWeek = (): void => {
+  const focusStartOfWeek = (): void => {
     if (!focusedDay) return;
-    const dayToFocus = addDays(
-      startOfWeek(addDays(focusedDay, -1), {
-        weekStartsOn: 0
-      }),
-      1
-    );
+    const dayToFocus = startOfWeek(focusedDay);
     switchMonth(dayToFocus, numberOfMonths);
     focus(dayToFocus);
   };
 
-  const focusLastDayOfWeek = (): void => {
+  const focusEndOfWeek = (): void => {
     if (!focusedDay) return;
-    const dayToFocus = addDays(
-      endOfWeek(addDays(focusedDay, -1), {
-        weekStartsOn: 0
-      }),
-      1
-    );
+    const dayToFocus = endOfWeek(focusedDay);
     switchMonth(dayToFocus, numberOfMonths);
     focus(dayToFocus);
   };
 
   const focusMonthBefore = (): void => {
     if (!focusedDay) return;
 
-    const month = subMonths(focusedDay, 1);
-    switchMonth(month, numberOfMonths);
-    const switchDay = Boolean(getDaysInMonth(month) < focusedDay.getDate());
-
-    focus(
-      !switchDay
-        ? month
-        : addDays(
-            startOfWeek(addWeeks(startOfMonth(month), 3)),
-            focusedDay.getDay()
-          )
```

Comment:
Why adding 3 weeks and get the start of the week of that?

Reply:
In case the day focused is 31th/30th or 29th - if that date does not exist in the prev/following month - it makes it to focus on the same day of the week

Reply:
otherwise the focus will disappear